spec/lib/aliquot/validator_spec.rb

```diff
@@ -0,0 +1,491 @@
+      end
+
+      it 'must be a string' do
+        generator.message_expiration = 123
+        is_expected.to eq({messageExpiration: ['must be a string']})
+      end
+
+      it 'must be an integer string' do
+        generator.message_expiration = 'not integer string'
+        is_expected.to eq({messageExpiration: ['must be string encoded integer']})
+      end
+    end
+
+    context 'messageId' do
+      it 'must exist' do
+        input.delete('messageId')
+        is_expected.to eq({messageId: ['is missing']})
+      end
+
+      it 'must be filled' do
+        generator.message_id = ''
+        is_expected.to eq({messageId: ['must be filled']})
+      end
+
+      it 'must be a string' do
+        generator.message_id = 123
+        is_expected.to eq({messageId: ['must be a string']})
+      end
+    end
+
+    context 'paymentMethod' do
+      it 'must exist' do
+        input.delete('paymentMethod')
+        is_expected.to eq({paymentMethod: ['is missing']})
+      end
+
+      it 'must be filled' do
+        generator.payment_method = ''
+        is_expected.to eq({paymentMethod: ['must be filled']})
+      end
+
+      it 'must be a string' do
+        generator.payment_method = 123
+        is_expected.to eq({paymentMethod: ['must be a string']})
+      end
+
+      it 'must be CARD' do
+        generator.payment_method = 'RANDOM'
+        is_expected.to eq({paymentMethod: ['must be equal to CARD']})
+      end
+    end
+
+    context 'paymentMethodDetails' do
+      it 'must exist' do
+        input.delete('paymentMethodDetails')
+        is_expected.to eq({paymentMethodDetails: ['is missing']})
+      end
+
+      it 'must be a JSON object' do
+        generator.payment_method_details = 'not a json object'
+        is_expected.to eq({paymentMethodDetails: ['must be a JSON object']})
+      end
+    end
+  end
+end
+
+describe Aliquot::Validator do
+  context 'ECv1' do
+    let(:generator) { AliquotPay.new(:ECv1) }
+    let(:token)     { generator.token }
+    subject do
+      schema.call(input).errors
+    end
+
+    include_examples 'Validator Spec'
+  end
+
+  context 'ECv2' do
+    let(:generator) { AliquotPay.new(:ECv2) }
+    let(:token)     { generator.token }
+    subject do
+      schema.call(input).errors
+    end
+    include_examples 'Validator Spec'
+
+    context 'SignedKeySchema' do
+      let(:schema) { Aliquot::Validator::SignedKeySchema }
+      let(:input)  { generator.build_signed_key }
+
+      context 'keyExpiration' do
+        it 'must exist' do
+          input.delete('keyExpiration')
+          is_expected.to eq({keyExpiration: ['is missing']})
+        end
+
+        it 'must be filled' do
+          input['keyExpiration'] = ''
+          is_expected.to eq({keyExpiration: ['must be filled']})
+        end
+
+        it 'must be integer string' do
+          generator.key_expiration = 'not digits'
+          is_expected.to eq({keyExpiration: ['must be string encoded integer']})
```

Comment:
The error message probably ought to be `must be string-encoded integer` :bike: :house: